test/hotspot/jtreg/compiler/predicates/assertion/TestAssertionPredicates.java

```diff
@@ -0,0 +1,471 @@
+ */
+
+/*
+ * @test id=StressXbatch
+ * @bug 8288981 8350579
+ * @requires vm.compiler2.enabled
+ * @run main/othervm -Xbatch -XX:+UnlockDiagnosticVMOptions -XX:+StressIGVN -XX:+StressGCM -XX:+AbortVMOnCompilationFailure
+ *                   -XX:CompileCommand=compileonly,compiler.predicates.TestAssertionPredicates::*
+ *                   -XX:CompileCommand=dontinline,compiler.predicates.TestAssertionPredicates::*
+ *                   compiler.predicates.TestAssertionPredicates Stress
+ */
+
+/*
+ * @test id=NoLoopPredication
+ * @bug 8288981 8350579
+ * @requires vm.compiler2.enabled
+ * @run main/othervm -Xbatch -XX:-UseLoopPredicate
+ *                   -XX:+UnlockDiagnosticVMOptions -XX:+AbortVMOnCompilationFailure
+ *                   -XX:CompileCommand=compileonly,compiler.predicates.TestAssertionPredicates::*
+ *                   -XX:CompileCommand=dontinline,compiler.predicates.TestAssertionPredicates::*
+ *                   compiler.predicates.TestAssertionPredicates NoLoopPredication
+ */
+
+/*
+ * @test id=NoFlags
+ * @bug 8288981 8350579
+ * @run main/othervm -XX:+UnlockDiagnosticVMOptions -XX:+AbortVMOnCompilationFailure
+ *                    compiler.predicates.TestAssertionPredicates NoFlags
+ */
+
+package compiler.predicates;
+
+public class TestAssertionPredicates {
+    static int[] iArrShort = new int[10];
+    static int[] iArr = new int[200];
+    static int[] iArr2 = new int[200];
+    static int[] iArr3 = new int[300];
+    static short[] sArr = new short[10];
+
+    static boolean flag;
+    static int iFld = 34;
+    static int iFld2, iFld3;
+    static float fFld;
+    static short five = 5;
+
+
+    public static void main(String[] args) {
+        switch (args[0]) {
+            case "Xbatch" -> {
+                for (int i = 0; i < 10000; i++) {
+                    flag = !flag;
+                    testTemplateAssertionPredicateNotRemovedHalt();
+                    testTemplateAssertionPredicateNotRemovedMalformedGraph();
+                    test8305428();
+                    test8305428No2();
+                }
+            }
+            case "NoTieredCompilation" -> {
+                for (int i = 0; i < 1000; i++) {
+                    test8320920();
+                    test8332501();
+                }
+            }
+            case "LoopMaxUnroll0" -> {
+                testDyingRuntimePredicate();
+                testDyingNegatedRuntimePredicate();
+            }
+            case "Xcomp" -> {
+                testDyingInitializedAssertionPredicate();
+            }
+            case "NoLoopPredication", "NoFlags", "Stress" -> {
+                for (int i = 0; i < 10000; i++) {
+                    runAllTests();
+                }
+            }
+            default -> throw new RuntimeException("invalid arg");
+        }
+    }
+
+    // Runs most of the tests except the really time-consuming ones.
+    static void runAllTests() {
```

Comment:
Sounds like a bit of a contradiction :sweat_smile: 

`runAllTests` -> `runAllFastTests`?

Reply:
Which ones are the really time-consuming ones? And why do you not run them here?

Reply:
I removed the comment for now - we run all tests here. This only applied to the full test from which I extracted these bits. I can revisit this comment and method name there again :-)